test/lib/resources/cdn_test.rb

```diff
@@ -0,0 +1,26 @@
+require 'katello_test_helper'
+class CdnResourceTest < ActiveSupport::TestCase
+  def test_http_downloader_v2
+    cdn_resource = Katello::Resources::CDN::CdnResource.new('http://foo.com')
+    Setting[:cdn_ssl_version] = 'SSLv23'
+    assert_equal cdn_resource.http_downloader.ssl_version, 'SSLv23'
+  end
+
+  def test_http_downloader_tls
+    cdn_resource = Katello::Resources::CDN::CdnResource.new('http://foo.com')
+    Setting[:cdn_ssl_version] = 'TLSv1'
```

Comment:
issue with the failing tests is that the Setting must be set before creating the cdn_resource